pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,178 @@
+package analysis
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	graphapi "github.com/gonum/graph"
+	"github.com/gonum/graph/path"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/api/kubegraph"
+	kubenodes "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph"
+	deploynodes "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	// HPAMissingScaleRefError denotes an error where a Horizontal Pod Autoscaler does not have a reference to an object to scale
+	HPAMissingScaleRefError = "HPAMissingScaleRef"
+	// HPAMissingCPUTargetError denotes an error where a Horizontal Pod Autoscaler does not have a CPU target to scale by.
+	// Currently, the only supported scale metric is CPU utilization, so without this metric an HPA is useless.
+	HPAMissingCPUTargetError = "HPAMissingCPUTarget"
+	// HPAOverlappingScaleRefWarning denotes a warning where a Horizontal Pod Autoscaler scales an object that is scaled by some other object as well
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+// FindHPASpecsMissingCPUTargets scans the graph in search of HorizontalPodAutoscalers that are missing a CPU utilization target.
+// As of right now, the only metric that HPAs can use to scale pods is the CPU utilization, so if a HPA is missing this target it
+// is effectively useless.
+func FindHPASpecsMissingCPUTargets(graph osgraph.Graph, namer osgraph.Namer, patchCommandName string) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastNode := range graph.NodesByKind(kubenodes.HorizontalPodAutoscalerNodeKind) {
+		node := uncastNode.(*kubenodes.HorizontalPodAutoscalerNode)
+
+		hpa, ok := node.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			continue
+		}
+
+		if hpa.Spec.CPUUtilization == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:       node,
+				Severity:   osgraph.ErrorSeverity,
+				Key:        HPAMissingCPUTargetError,
+				Message:    fmt.Sprintf("%s is missing a CPU utilization target", namer.ResourceName(node)),
+				Suggestion: osgraph.Suggestion(fmt.Sprintf(`%s %s --patch '{"spec":{"cpuUtilization":{"targetPercentage": 80}}}'`, patchCommandName, namer.ResourceName(node))),
```

Comment:
> Hm. I'd almost argue for a user who is not knowlegeable, the longer flags are better as they are more descriptive of what is happening.

Stay consistent with the others.  We've chosen short flags and short names to keep the text tight and `-p` isn't very ambiguous in this context.